clcache.py

```diff
@@ -448,15 +451,18 @@ def save(self):
 class AnalysisResult:
     Ok, NoSourceFile, MultipleSourceFilesSimple, \
         MultipleSourceFilesComplex, CalledForLink, \
-        CalledWithPch, ExternalDebugInfo = range(7)
+        CalledWithPch, ExternalDebugInfo = list(range(7))
 
 
-def getFileHash(filePath, additionalData=None):
+def getFileHash(filePath, additionalDataString=None):
     hasher = HASH_ALGORITHM()
     with open(filePath, 'rb') as inFile:
         hasher.update(inFile.read())
-    if additionalData is not None:
-        hasher.update(additionalData)
+    if additionalDataString is not None:
+        # Encoding of this additional data does not really matter
+        # as long as we keep it fixed, otherwise hashes change.
+        # The string should fit into ASCII, so UTF8 should not change anything
+        hasher.update(additionalDataString.encode("UTF-8"))
```

Comment:
`additionalData` is a string, i.e. a list of unicode code points. Those unicode strings are not a sequence of bytes but the hasher can only consume byte sequences. There are multiple ways to "encode" a unicode string into a sequence of bytes, for example latin1, UTF-8 or UTF-16. All of them will generate different binary representations for the same string. Python 2 does not strictly differentiate between unicode strings and byte sequences, but Python 3 does.

Without explicitly encoding, Python 3 tell us exactly that:

```
File "[...]\clcache.py", line 465, in getFileHash
    hasher.update(additionalDataString)
TypeError: Unicode-objects must be encoded before hashing
```